backend/src/packages/drivers/driver.service.ts

```diff
@@ -0,0 +1,157 @@
+    driverRepository: DriverRepository,
+    userService: UserService,
+    groupService: GroupService,
+  ) {
+    this.driverRepository = driverRepository;
+    this.userService = userService;
+    this.groupService = groupService;
+  }
+
+  public async findById(id: number): Promise<DriverEntityT | null> {
+    const [driver = null] = await this.driverRepository.find({ id });
+
+    return driver ? DriverEntity.initialize(driver).toObject() : null;
+  }
+
+  public async findAllByBusinessId(
+    businessId: number,
+  ): Promise<DriverGetAllResponseDto> {
+    const items = await this.driverRepository.findAllByBusinessId(businessId);
+
+    return {
+      items: items.map((it) => it.toObject()),
+    };
+  }
+
+  public async create({
+    payload,
+    id,
+  }: DriverCreatePayload): Promise<DriverAddResponseWithGroup> {
+    const { password, email, lastName, firstName, phone, driverLicenseNumber } =
+      payload;
+
+    const { result: doesDriverExist } = await this.driverRepository.checkExists(
+      {
+        driverLicenseNumber,
+      },
+    );
+
+    if (doesDriverExist) {
+      throw new HttpError({
+        status: HttpCode.BAD_REQUEST,
+        message: HttpMessage.DRIVER_ALREADY_EXISTS,
+      });
+    }
+    const group = await this.groupService.findByKey(UserGroupKey.DRIVER);
+
+    if (!group) {
+      throw new HttpError({
+        message: HttpMessage.INVALID_GROUP,
+        status: HttpCode.BAD_REQUEST,
+      });
+    }
+
+    const user = await this.userService.create({
+      password,
+      email,
+      lastName,
+      firstName,
+      phone,
+      groupId: group.id,
+    });
+
+    const driver = await this.driverRepository.create(
+      DriverEntity.initializeNew({
+        driverLicenseNumber,
+        businessId: id,
+        userId: user.id,
+      }),
+    );
+
+    const driverObject = driver.toObject();
+
+    return { ...user, group, driver: driverObject };
+  }
+
+  public async update({
+    driverId,
+    payload,
+  }: DriverUpdatePayload): Promise<DriverUpdateResponseDto> {
+    const foundDriver = await this.findById(driverId);
+
+    if (!foundDriver) {
+      throw new NotFoundError({});
+    }
+
+    const { result: doesDriverExist } = await this.driverRepository.checkExists(
+      {
+        driverLicenseNumber: payload.driverLicenseNumber,
+      },
+    );
+
+    if (doesDriverExist) {
+      throw new HttpError({
+        status: HttpCode.BAD_REQUEST,
+        message: HttpMessage.LICENSE_NUMBER_ALREADY_EXISTS,
+      });
+    }
+
+    const user = await this.userService.update(driverId, payload);
+
+    const driver = await this.driverRepository.update({
+      id: foundDriver.id,
+      payload,
+    });
+
+    const driverObject = driver.toObject();
+
+    return { ...user, driver: driverObject };
+  }
+
+  public async delete(id: number): Promise<boolean> {
+    const foundDriver = await this.findById(id);
+
+    if (!foundDriver) {
+      throw new NotFoundError({});
```

Comment:
As I understand it will throw 500 error, may be should use HttpError

Reply:
This is our custom error for Not Found, which has the path ~libs/exceptions/not-found-error. So no, it doesn't throw a 500 error. The only thing that might be worth adding is a different message.

Reply:
Alex is right, maybe you should use it like this: 
 `throw new HttpError({
        status
        message
      });` In this case, you can't see 500.